go/arrow/ipc/file_reader.go

```diff
@@ -617,6 +626,45 @@ func (ctx *arrayLoaderContext) loadStruct(dt *arrow.StructType) arrow.ArrayData
 	return array.NewData(dt, int(field.Length()), buffers, subs, int(field.NullCount()), 0)
 }
 
+func (ctx *arrayLoaderContext) loadUnion(dt arrow.UnionType) arrow.ArrayData {
+	nBuffers := 2
```

Comment:
Maybe a small comment on what 2 and 3 represent here.

Reply:
added